### PR TITLE
HDDS-2793. concept/Datanodes.md translation

### PR DESCRIPTION
**Pull request by iamabug**
## What changes were proposed in this pull request?

Datanodes.zh.md added

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-2793

## How was this patch tested?

hugo server -D

### CODE REVIEW

File: hadoop-hdds/docs/content/concept/Datanodes.zh.md [hunk 1]
```diff
@@ -21,4 +21,31 @@ summary: TODO translated summary
...
+当用户定位到容器的位置，即知道哪些数据节点包含这个容器后，用户会连接数据节点，然后读取由 _容器ID:本地ID_ 指定的数据流，换句话说，本地 ID 相当于容器内的索引，描述了我们应该读取哪个数据流。
+
+### 容器的位置发现
+
+SCM 如何获得容器的位置？方法和 HDFS 中的方法十分相似。数据节点会定期发送类似于块报告的容器报告，但比块报告的内容简洁的多，比如，对于一个存储容量为 196 TB 的单数据节点组成的集群，Ozone 大概会拥有四万个容器，相比于 HDFS 的一百五十万个块，块报告数量缩减为四十分之一。
+
+这种间接管理的方式大大地提高了 Ozone 的扩展性，SCM 需要处理的块数据大大减少，TODO
+This extra indirection helps tremendously with scaling Ozone. SCM has far
+less block data to process and the name node is a different service are
+critical to scaling Ozone.
```
Comment (iamabug) [inline hadoop-hdds/docs/content/concept/Datanodes.zh.md]: `the name node is a different service are critical to scaling Ozone`

I think it does not make sense in grammar.

Comment (xiaoyuyao) [reply hadoop-hdds/docs/content/concept/Datanodes.zh.md]: Good catch. Can you change EN document from  "name node is a" to " namespace service (Ozone Manager) as a "? That will make sense.

File: hadoop-hdds/docs/content/concept/Datanodes.zh.md [hunk 2]
```diff
@@ -1,8 +1,8 @@
 ---
-title: "Datanodes"
+title: "数据节点"
 date: "2017-09-14"
 weight: 4
-summary: TODO translated summary
+summary: Ozone 支持 Amazon 的简易存储服务（S3）协议，你可以原封不动地在 Ozone 上使用基于 S3 客户端和 S3 SDK 的应用。
```
Comment (xiaoyuyao) [inline hadoop-hdds/docs/content/concept/Datanodes.zh.md]: Amazon 的简易存储服务（S3）=> Amazon S3

File: hadoop-hdds/docs/content/concept/Datanodes.zh.md [hunk 3]
```diff
@@ -21,4 +21,31 @@ summary: TODO translated summary
...
 
-TODO: content translations
\ No newline at end of file
+数据节点是 Ozone 中的 worker，所有的数据都存储在数据节点上，用户以块的方式写数据，数据节点将多个块聚合成一个存储容器，存储容器中包含用户写入的数据块和这些块的元数据。
+
+## 存储容器
+
+![FunctionalOzone](ContainerMetadata.png)
+
+Ozone 的存储容器是一个自包含的超级块，容器中包含一系列的 Ozone 块，以及存储实际数据的磁盘文件，这是默认的存储容器格式。对于 Ozone 来说，容器只是提供了一个协议说明，实际的存储方式并不重要，换句话说，扩展或引入新的容器设计也是很容易的。因此，上述格式应当被看作是 Ozone 存储容器的参考实现。
```
Comment (xiaoyuyao) [inline hadoop-hdds/docs/content/concept/Datanodes.zh.md]: 协议说明=>协议规范

Comment (xiaoyuyao) [reply hadoop-hdds/docs/content/concept/Datanodes.zh.md]: 实际的存储方式并不重要=》它独立于具体的存储格式实现。

Comment (xiaoyuyao) [reply hadoop-hdds/docs/content/concept/Datanodes.zh.md]: 扩展或引入新的容器设计也是很容易的=》我们可以很容易扩展或引入新的容器实现格式。

File: hadoop-hdds/docs/content/concept/Datanodes.zh.md [hunk 4]
```diff
@@ -21,4 +21,31 @@ summary: TODO translated summary
...
+
+![OzoneBlock](OzoneBlock.png)
+
+容器 ID 用来让用户发现容器的位置，容器位置的权威信息存储在 SCM 中。大部分情况下，OM 会缓存容器的位置信息，并会将它们随 Ozone 块一起返回给用户。
+
+当用户定位到容器的位置，即知道哪些数据节点包含这个容器后，用户会连接数据节点，然后读取由 _容器ID:本地ID_ 指定的数据流，换句话说，本地 ID 相当于容器内的索引，描述了我们应该读取哪个数据流。
+
+### 容器的位置发现
+
+SCM 如何获得容器的位置？方法和 HDFS 中的方法十分相似。数据节点会定期发送类似于块报告的容器报告，但比块报告的内容简洁的多，比如，对于一个存储容量为 196 TB 的单数据节点组成的集群，Ozone 大概会拥有四万个容器，相比于 HDFS 的一百五十万个块，块报告数量缩减为四十分之一。
```
Comment (xiaoyuyao) [inline hadoop-hdds/docs/content/concept/Datanodes.zh.md]: 方法和 HDFS 中的方法十分相似=》这一点和现有的HDFS 十分相似

Comment (xiaoyuyao) [reply hadoop-hdds/docs/content/concept/Datanodes.zh.md]: 但比块报告的内容简洁的多=》容器报告比块报告的内容简洁的多

Comment (xiaoyuyao) [reply hadoop-hdds/docs/content/concept/Datanodes.zh.md]: 196 TB 的单数据节点组成的集群-》196 TB 的集群

**Comment by iamabug** [comment]
retest

**Comment by xiaoyuyao** [comment]
I check the acceptance test logs which points to the om-ha tests with permission issue. This does not seem to relate to the document change here.  cc: @hanishakoneru and @bharatviswa504  


2019-11-16T11:59:00.0117168Z ==============================================================================
2019-11-16T11:59:00.0120741Z ozone-om-ha-testOMHA :: Smoketest ozone cluster startup                       
2019-11-16T11:59:00.0121273Z ==============================================================================
2019-11-16T11:59:12.8359463Z Stop Leader OM and Verify Failover                                    | FAIL |
2019-11-16T11:59:12.8360383Z OSError: [Errno 13] Permission denied: '/opt/hadoop/tmpsCDIs5'
2019-11-16T11:59:12.8360708Z ------------------------------------------------------------------------------
2019-11-16T12:01:07.8768425Z Restart OM and Verify Ratis Logs                                      | PASS |
2019-11-16T12:01:07.8772261Z ------------------------------------------------------------------------------
2019-11-16T12:01:24.9714368Z Test Multiple Failovers                                               | FAIL |
2019-11-16T12:01:24.9715403Z OSError: [Errno 13] Permission denied: '/opt/hadoop/tmpLn9PKJ'
2019-11-16T12:01:24.9715697Z ------------------------------------------------------------------------------
2019-11-16T12:01:24.9742716Z ozone-om-ha-testOMHA :: Smoketest ozone cluster startup               | FAIL |
2019-11-16T12:01:24.9749529Z 3 critical tests, 1 passed, 2 failed
2019-11-16T12:01:24.9749662Z 3 tests total, 1 passed, 2 failed

**Comment by hanishakoneru** [comment]
@xiaoyuyao, OM-HA tests are disabled and should not be running. 
In the following CI run, the acceptance test suite does not run OM-HA tests:
https://github.com/apache/hadoop-ozone/pull/520/checks?check_run_id=485882139

Is there any other run you are referring to?

**Comment by xiaoyuyao** [comment]
If you download the acceptance test results from the link and 
https://github.com/apache/hadoop-ozone/suites/485213917/artifacts/2531797

search for this file 7_Run .hadoop-ozonedev-supportchecksacceptance.sh.txt
You can find the error above.

**Comment by xiaoyuyao** [comment]
==============================================================================
2019-11-16T11:59:00.0120726Z ozone-om-ha-testOMHA :: Smoketest ozone cluster startup                       
2019-11-16T11:59:00.0121265Z ==============================================================================
2019-11-16T11:59:12.8359426Z Stop Leader OM and Verify Failover                                    | FAIL |
2019-11-16T11:59:12.8360374Z OSError: [Errno 13] Permission denied: '/opt/hadoop/tmpsCDIs5'
2019-11-16T11:59:12.8360702Z ------------------------------------------------------------------------------
2019-11-16T12:01:07.8768389Z Restart OM and Verify Ratis Logs                                      | PASS |
2019-11-16T12:01:07.8772219Z ------------------------------------------------------------------------------
2019-11-16T12:01:24.9714328Z Test Multiple Failovers                                               | FAIL |
2019-11-16T12:01:24.9715394Z OSError: [Errno 13] Permission denied: '/opt/hadoop/tmpLn9PKJ'
2019-11-16T12:01:24.9715691Z

**Comment by hanishakoneru** [comment]
This error is from an old run. Confirmed with @xiaoyuyao that the new runs do not have the OM HA tests.

**Comment by adoroszlai** [comment]
Thanks @iamabug for the translation and @xiaoyuyao for reviewing it.

**Pull request merged**